frontend/catalyst/jax_extras/tracing.py

```diff
@@ -248,6 +312,97 @@ def __init__(self, boxid: int, e: JaxprEqn):
     return [b.e for b in stable_toposort(boxes)]  # [4]
 
 
+@transformation_with_aux
+def expanded_fun(static_args, *args_expanded):
+    """Function transformation making the function to accept its arguments in the expanded format
+    (with the dimension variables added)"""
```

Comment:
Could you please clarify the question? In general, this transform turns a list of fully expanded arguments into a regular Python arguments, passes it to a user-defined function, and records `OutputType` (as Jax defines it) and return it along with jaxpr and constants. Since user-facing function needs collapsed list of arguments, this transform actually removes the implicit arguments from the list.